rotkehlchen/exchanges/bitstamp.py

```diff
@@ -721,3 +728,40 @@ def _process_unsuccessful_response(
             return []
 
         raise AssertionError(f'Unexpected Bitstamp response_case: {case}.')
+
+    def _query_exchange_pairs(self) -> ExchangePairsResponse:
+        """Query the exchange pairs data and return the list of names in
+        `<ExchangePairsResponse>.pairs`.
+        Otherwise populate <ExchangePairsResponse> with data that each endpoint
+        can process as an unsuccessful request.
+        """
+        was_successful = True
+        pairs = []
+        request_url = f'{self.base_uri}/v2/trading-pairs-info'
+        log.debug(f'{self.name} API request', request_url=request_url)
+        try:
+            response = self.session.get(request_url)
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(
+                f'{self.name} get request at {request_url} connection error: {str(e)}.',
+            ) from e
+
+        if response.status_code != HTTPStatus.OK:
```

Comment:
General comment which I saw you also do a lot in Bitfinex.

Please try to return early for errors. Not big if /elses like this that can go into many indentation levels. I know you do it so you can return from one final place at the end but in bigger functions it becomes unreadable.

In general try to return early. This function is too small to care, and you don't need to change it. But I would like you to follow the return early principle: https://medium.com/swlh/return-early-pattern-3d18a41bba8

Reply:
I will read it, thanks!